enhancements/api-review/gcp_user_defined_labels.md

```diff
@@ -0,0 +1,305 @@
+                          type: object
+                          required:
+                            - key
+                            - value
+                          properties:
+                            key:
+                              description: key is the name of the label
+                              type: string
+                              maxLength: 63
+                              minLength: 1
+                              pattern: ^[a-z][0-9a-z_-]+$
+                            value:
+                              description: value is the value of the label.
+                              type: string
+                              maxLength: 63
+                              minLength: 1
+                              pattern: ^[0-9a-z_-]+$
+```
+
+### Implementation Details/Notes/Constraints [optional]
+Add a new field `resourceLabels` to `.status.platformStatus.gcp` of the 
+`infrastructure.config.openshift.io` type. Labels included in the `resourceLabels` field 
+will be applied to new resources created for the cluster by the in-cluster operators.
+
+The `resourceLabels` field in `status.platformStatus.gcp` will be populated by the 
+installer using the entries from `platform.gcp.userLabels` field of `install-config`.
+
+`status.platformStatus.gcp` field of `infrastructure.config.openshift.io` is immutable.
+
+All operators that create GCP resources (Cluster Infrastructure ,Storage ,Node ,
+NetworkEdge ,Internal Registry ,CCO) will apply these labels to all GCP resources they create.
+
+Below list of terraform GCP APIs to create resources should be updated to add user
+defined labels and as well the openshift default label in the installer component.
+`google_storage_bucket, google_compute_instance, google_dns_managed_zone, google_compute_image`
+
+And below list of terraform GCP APIs support labeling resources in beta version.
+`google_compute_address, google_compute_forwarding_rule`
+
+Below list of GCP Resources does not support labeling.
+`VPC Network, VM Instance Group, Subnetwork, Target Pool, DNS Zone, Records, Router, NAT Rules, Backend Service, Health Check`
+
+API update example:
+A local variable should be defined, which merges the default label and the user
+defined GCP labels, which should be referred in the GCP resource APIs.
+``` terraform
+locals {
+  labels = merge(
+    {
+      "kubernetes.io_cluster.${var.cluster_id}" = "owned"
+    },
+    var.gcp_extra_labels,
+  )
+}
+
+resource "google_storage_bucket" "ignition" {
+  ...
+  labels = local.labels
+}
+```
+
+For resources supporting labels in beta version below changes are required
+``` terraform
+provider "googlebeta" {
+  credentials = var.gcp_service_account
+  project     = var.gcp_project_id
+  region      = var.gcp_region
+}
+
+resource "google_compute_address" "cluster_ip" {
+  provider = googlebeta
+  ...
+  labels = local.labels
+}
+```
```

Comment:
we don't currently use the googlebeta provider. We would need to have a discussion about whether there are any downsides to using that provider and whether those resources merit any disadvantages

Reply:
Thank you!! A discussion will be helpful, before asking customers if those two resources need to have labels and if yes what are the issues of using googlebeta.

Reply:
By discussion, I mean if you could just include in the enhancement some details about the googlebeta provider. It seems like what would need to happen is we would take existing resources that are being created with the normal provider and would switch those to using [the googlebeta provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_versions). The kinds of questions we would want to explore are:

1. Would switching existing resources to a different provider have any noticeable effects?
2. In particular, would customers need to enable additional APIs/services to use this provider.

Reply:
Updated `Risks and Mitigations` section with more details on google-beta provider.